File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/rxRepository/rxDatasource/OcmDiskDataStore.java
package com.gigigo.orchextra.core.data.rxRepository.rxDatasource;

import android.content.Context;
import com.gigigo.orchextra.core.data.api.dto.content.ApiSectionContentData;
import com.gigigo.orchextra.core.data.api.dto.elements.ApiElementData;
import com.gigigo.orchextra.core.data.api.dto.menus.ApiMenuContentData;
import com.gigigo.orchextra.core.data.api.dto.versioning.ApiVersionKache;
import com.gigigo.orchextra.core.data.api.dto.video.ApiVideoData;
import com.gigigo.orchextra.core.data.rxCache.OcmCache;
import io.reactivex.Observable;
import orchextra.javax.inject.Inject;
import orchextra.javax.inject.Singleton;

@Singleton public class OcmDiskDataStore implements OcmDataStore {
  private final OcmCache ocmCache;

  @Inject public OcmDiskDataStore(OcmCache ocmCache) {
    this.ocmCache = ocmCache;
  }

  @Override public Observable<ApiMenuContentData> getMenuEntity() {
    return ocmCache.getMenus().doOnNext(apiMenuContentData -> {
      apiMenuContentData.setFromCloud(false);
    });
  }

  @Override public Observable<ApiSectionContentData> getSectionEntity(String elementUrl,
      int numberOfElementsToDownload) {
    return ocmCache.getSection(elementUrl).doOnNext(apiSectionContentData -> {
      apiSectionContentData.setFromCloud(false);
    });
  }

  @Override public Observable<ApiSectionContentData> searchByText(String section) {
    return null;
  }

  @Override public Observable<ApiElementData> getElementById(String slug) {
    return ocmCache.getDetail(slug);
  }

  @Override public Observable<ApiVideoData> getVideoById(Context context, String videoId,
      boolean isWifiConnection, boolean isFastConnection) {
    return ocmCache.getVideo(videoId);
  }

  @Override public Observable<ApiVersionKache> getVersion() {
    return ocmCache.getVersion();
  }

  @Override public boolean isFromCloud() {
    return false;
  }

  public OcmCache getOcmCache() {
    return ocmCache;
  }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/rxRepository/rxDatasource/OcmDiskDataStore.kt
package com.gigigo.orchextra.core.data.rxRepository.rxDatasource

import android.content.Context
import com.gigigo.orchextra.core.data.database.entities.DbElementData
import com.gigigo.orchextra.core.data.database.entities.DbMenuContentData
import com.gigigo.orchextra.core.data.database.entities.DbSectionContentData
import com.gigigo.orchextra.core.data.database.entities.DbVideoData
import com.gigigo.orchextra.core.data.mappers.toContentData
import com.gigigo.orchextra.core.data.mappers.toElementData
import com.gigigo.orchextra.core.data.mappers.toMenuContentData
import com.gigigo.orchextra.core.data.mappers.toVimeoInfo
import com.gigigo.orchextra.core.data.rxCache.OcmCache
import com.gigigo.orchextra.core.domain.entities.contentdata.ContentData
import com.gigigo.orchextra.core.domain.entities.elements.ElementData
import com.gigigo.orchextra.core.domain.entities.menus.MenuContentData
import gigigo.com.vimeolibs.VimeoInfo
import io.reactivex.Observable
import orchextra.javax.inject.Inject
import orchextra.javax.inject.Singleton

@Singleton
class OcmDiskDataStore @Inject constructor(val ocmCache: OcmCache) : OcmDataStore {

  override fun getMenus(): Observable<MenuContentData> {
    return ocmCache.getMenus().map(DbMenuContentData::toMenuContentData)
  }

  override fun getSection(elementUrl: String,
      numberOfElementsToDownload: Int): Observable<ContentData> {
    return ocmCache.getSection(elementUrl).map(DbSectionContentData::toContentData)
  }

  override fun searchByText(section: String): Observable<ContentData>? {
    return null
  }

  override fun getElementById(slug: String): Observable<ElementData> {
    return ocmCache.getDetail(slug).map(DbElementData::toElementData)
  }

  override fun getVideoById(context: Context, videoId: String, isWifiConnection: Boolean,
      isFastConnection: Boolean): Observable<VimeoInfo> {
    return ocmCache.getVideo(videoId).map(DbVideoData::toVimeoInfo)
  }

  override fun isFromCloud(): Boolean {
    return false
  }
}